src/Exceptions/AuthenticationException.php

```diff
@@ -11,6 +11,17 @@ public function toResponse($request)
     {
         return $request->expectsJson()
             ? response()->json(['message' => $this->getMessage()], 401)
-            : redirect()->route('statamic.cp.login');
+            : $this->handleRedirect();
+    }
+
+    protected function handleRedirect()
+    {
+        if (! config('statamic.cp.auth.enabled', true)) {
+            return config('statamic.cp.auth.redirect_to')
+                ? redirect(config('statamic.cp.auth.redirect_to'))
```

Comment:
this should be `Redirect::guest(config('statamic.cp.auth.redirect_to'))` so that the `url.intended` is set

Reply:
(and of course there should be a test for that, I'll try to come up with one as well and submit this, if I have time.

Reply:
I don't know if we actually have any existing tests for the CP authentication stuff - we only have tests for the frontend auth tags.